Source/Core/Common/x64Emitter.cpp

```diff
@@ -1540,6 +1540,13 @@ void XEmitter::XOR(int bits, const OpArg& a1, const OpArg& a2)
 }
 void XEmitter::MOV(int bits, const OpArg& a1, const OpArg& a2)
 {
+  // Shortcut to zero a register
+  if (a2.IsZero() && a1.IsSimpleReg() && !flags_locked)
+  {
+    XOR(bits, a1, a1);
+    return;
+  }
```

Comment:
Can I propose reverting this change, and moving the logic to a higher level wrapper function (say, "move")? There's no longer a simple way to generate "mov eax, 0", and the rest of the code assumes (pretty fairly) that emitting a "MOV" won't modify flags.

In particular, I think this introduces errors in [Jit64::arithXex](https://github.com/dolphin-emu/dolphin/blob/93a5efa347c62ca97e375a1a226d7741d584fd2f/Source/Core/Core/PowerPC/Jit64/Jit_Integer.cpp#L1311) which first unlocks the flags (line 1328) then may emit MOV before emitting SBB or ADC. (Not sure if this relates to [issue 9661](https://dolp.in/i9661), but it seems plausible)